core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1385,13 +1391,11 @@ private Optional<RetentionSizeData> buildRetentionSizeData(long retentionSize,
                     Iterator<RemoteLogSegmentMetadata> segmentsIterator = remoteLogMetadataManager.listRemoteLogSegments(topicIdPartition, epoch);
                     while (segmentsIterator.hasNext()) {
                         RemoteLogSegmentMetadata segmentMetadata = segmentsIterator.next();
-                        // Only count the size of "COPY_SEGMENT_FINISHED" and "DELETE_SEGMENT_STARTED" state segments
-                        // because "COPY_SEGMENT_STARTED" means copy didn't complete, and "DELETE_SEGMENT_FINISHED" means delete did complete.
-                        // Note: there might be some "COPY_SEGMENT_STARTED" segments not counted here.
-                        // Either they are being copied and will be counted next time or they are dangling and will be cleaned elsewhere,
-                        // either way, this won't cause more segment deletion.
-                        if (segmentMetadata.state().equals(RemoteLogSegmentState.COPY_SEGMENT_FINISHED) ||
-                                segmentMetadata.state().equals(RemoteLogSegmentState.DELETE_SEGMENT_STARTED)) {
+                        // Count only the size of segments in "COPY_SEGMENT_FINISHED" state because 
+                        // "COPY_SEGMENT_STARTED" means copy didn't complete and we will count them later, 
+                        // "DELETE_SEGMENT_STARTED" means deletion failed and we will retry later,
```

Comment:
Now, we are triggering the deletion in 2 tasks: copyTask and expirationTask. Can it be updated to:

```
"DELETE_SEGMENT_STARTED" means deletion initiated (or) failed in the previous attempt and we will retry later,
```

Reply:
Sure.